src/webgl/p5.RendererGL.js

```diff
@@ -1033,16 +1034,46 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
       }
     }
 
-    // apply shader to pg
     pg.clear(); // prevent undesirable feedback effects accumulating secretly
-    pg.shader(this.filterShader);
-    this.filterShader.setUniform('tex0', this);
-    this.filterShader.setUniform('texelSize', [1.0/this.width, 1.0/this.height]);
-    this.filterShader.setUniform('canvasSize', [this.width, this.height]);
-    // filterParameter only used for POSTERIZE, BLUR, and THRESHOLD
-    // but shouldn't hurt to always set
-    this.filterShader.setUniform('filterParameter', filterParameter);
-    pg.rect(0,0,this.width,this.height);
+
+    // apply blur shader with multiple passes
+    if (operation === constants.BLUR) {
+
+      // pg is the accumulator. initialize with contents of main renderer (this)
+      pg.copy(
+        this,
+        0, 0, this.width, this.height,
+        -this.width/2, -this.height/2, this.width, this.height
+      );
+      // how much to blur, given by user
+      let steps = filterParameter;
+
+      for (let i = 0; i < steps; i++) {
+        // first pass averaging horizontal neighbors
+        pg.shader(this.filterShader);
+        this.filterShader.setUniform('texelSize', [1/this.width, 1/this.height]);
+        this.filterShader.setUniform('tex0', pg);
```

Comment:
Does this work? Aren't you drawing to the same buffer that you're writing to?

Reply:
It seems like it works now just using one renderer, but I was doing it wrong before. `setUniform('tex0', this)` seems to have helped.